### ISSUE TITLE
Result of async.timeout() only works the first time you call it.

### ISSUE THREAD

**Issue by acjohnso25** [caolan/async#1418]
**What version of async are you using?**  2.4.0

**Which environment did the issue occur in (Node version/browser version)**  Node.js 4.7.0

**What did you do? Please include a minimal reproducable case illustrating issue.**  
```
const async = require('async');

var asyncFn = function (i, done) {
    if (i < 1) {
        setTimeout(done, 200);
    } else {
        setImmediate(done);
    }
};

// wrap function once, call 10 times
var timeoutFn = async.timeout(asyncFn, 100);
var j = 0;
async.whilst(
    function () {
        return j < 10;
    },
    function (done) {
        timeoutFn(j++, function (err) {
            console.log('Test 1 #' + j, err);
            done();
        });
    },
    function (err) {
    }
);
```

**What did you expect to happen?**  Should only get timeout error the first time it's called.

**What was the actual result?**  Timeout occurs every time it's called, even though the underlying function only takes too long the first time.

**Comment by acjohnso25** [comment]
This may actually be by design, but if so it wasn't what I expected, and it wasn't indicated in the documentation from what I could see.

**Comment by hargasinski** [comment]
Hi @acjohnso25, I believe you're right that this is bug. It should only give a `ETIMEDOUT` error on the first call. Thanks for the report, we'll fix as soon as possible.

**Comment by acjohnso25** [comment]
I guess it's more correct to say that it works _until_ the first timeout...then it _always_ times out.  For example, if you change `asyncFn` to this:

```
var asyncFn = function (i, done) {
    if (i === 5) {
        setTimeout(done, 200);
    } else {
        setImmediate(done);
    }
};
```

...then what you see is:

```
Test 1 #1 undefined
Test 1 #2 undefined
Test 1 #3 undefined
Test 1 #4 undefined
Test 1 #5 undefined
Test 1 #6 { [Error: Callback function "anonymous" timed out.] code: 'ETIMEDOUT' }
Test 1 #7 { [Error: Callback function "anonymous" timed out.] code: 'ETIMEDOUT' }
Test 1 #8 { [Error: Callback function "anonymous" timed out.] code: 'ETIMEDOUT' }
Test 1 #9 { [Error: Callback function "anonymous" timed out.] code: 'ETIMEDOUT' }
Test 1 #10 { [Error: Callback function "anonymous" timed out.] code: 'ETIMEDOUT' }
```

**Comment by aearly** [comment]
Yeesh, I almost feel like we should do patch releases back to 2.0.0.

**Comment by aearly** [comment]
Fixed in v2.4.1
